python/katana/example_data.py

```diff
@@ -3,67 +3,127 @@
 """
 
 import os
-import shutil
-import tarfile
-import urllib.request
 from pathlib import Path
 
-__all__ = ["get_input", "get_input_as_url"]
+__all__ = ["get_rdg_dataset_at_version", "get_rdg_dataset"]
+
+## functions to access test datasets git repo
+def _as_url(path: str) -> str:
+    """
+    Return the path as file:// URL.
+    """
+    path = f"file://{path}"
+    return path
 
 
-def get_inputs_directory(*, invalidate=False) -> Path:
-    inputs_dir = None
+def _get_latest_rdg_storage_format_version() -> int:
+    latest_rdg_storage_format_version = os.environ.get(["KATANA_RDG_STORAGE_FORMAT_VERSION"], None)
+    if latest_rdg_storage_format_version is None:
+        raise ValueError("KATANA_RDG_STORAGE_FORMAT_VERSION must be set by cmake")
+    return latest_rdg_storage_format_version
 
-    # Use the build paths if they exist.
-    if "KATANA_BUILD_DIR" in os.environ:
-        # If KATANA_BUILD_DIR environment is set, just use it
-        paths_to_check = [Path(os.environ["KATANA_BUILD_DIR"])]
-    else:
+
+def _get_test_datasets_directory() -> Path:
+    """
+    Locate the test datasets directory
+    """
+    test_datasets_path = None
+    # If TEST_DATASETS environment is set, just use it
+    test_datasets_path = os.environ.get("TEST_DATASETS", None)
+    if test_datasets_path is None:
+        # else try to find it
         paths_to_check = list(Path(__file__).parents) + list(Path.cwd().parents)
-    for path in paths_to_check:
-        # TODO(amp): If we can abstract the input version info a shared file, this should look for
-        #  specifically that version.
-        ci_inputs_path = (path / "inputs" / "current").resolve()
-        if ci_inputs_path.is_dir():
-            inputs_dir = ci_inputs_path
-    # Otherwise use a cache directory
-    if not inputs_dir:
-        inputs_dir = Path(os.environ["HOME"]) / ".cache" / "katana" / "inputs"
-    if inputs_dir.is_dir() and (inputs_dir / "propertygraphs" / "ldbc_003").is_dir():
-        if not invalidate:
-            return inputs_dir
-        try:
-            shutil.rmtree(inputs_dir)
-        except OSError:
-            inputs_dir.unlink()
-    inputs_dir.mkdir(parents=True, exist_ok=True)
-    fn, _headers = urllib.request.urlretrieve(
-        "https://katana-ci-public.s3.us-east-1.amazonaws.com/inputs/katana-inputs-v30.tar.gz"
-    )
```

Comment:
I think you missed a critical feature of this code: It will download the inputs if the user doesn't have a build directory. This allowed users to get these dataset easily from within a notebook since the code would automatically download the data for them when they required it.

It should be possible do download the data if it's not available. Github provides zip files of repo content. In this case: https://github.com/KatanaGraph/test-datasets/archive/refs/heads/master.zip.

See main comment for a bit more discussion of this.

Reply:
Is this actually an issue since the git repo is now part of the source, rather than the build?

I'm not sure I understand why we would want to support someone cloning the katana repo without the test-datasets submodule?

(That said, I'm now sure I have to update the docs/README for downloading katana)

Reply:
I don't think it should be a problem since it should be a part of the source module now. 

On a tangent, I think the advantage of having the downloadable inputs was that you could always download the latest version of the dataset whereas now you have to update a submodule, which can be a bit of a pain to some. My hope is that as our RDG format changes stabilize this will become less of an issue.

Reply:
Talked to @arthurp out of band who said:

> The issue is that the datasets are not just for testing they are also for end user experimentation.
> The user will not have the source tree in hand.

so I will be addressing this

Reply:
alright, I have addressed this issue. It makes me sad that we must hardcode a sha to download, but the other option is to just have the code always pull test-datasets master, which could cause breakages in notebooks/users code. 
Theoretically, we could be smart and store the sha of test-datasets in use at build time, but I will leave that as a TODO.